lib/mini_scheduler/distributed_mutex.rb

```diff
@@ -18,11 +19,28 @@ def initialize(key, redis)
       @mutex = Mutex.new
     end
 
+    MAX_POLLING_ATTEMPTS ||= 60
+    BASE_SLEEP_DURATION ||= 0.001
+    MAX_SLEEP_DURATION ||= 1
+
     # NOTE wrapped in mutex to maintain its semantics
     def synchronize
       @mutex.lock
+
+      attempts = 0
+      sleep_duration = BASE_SLEEP_DURATION
       while !try_to_get_lock
-        sleep 0.001
+        sleep sleep_duration
+
+        # Exponential backoff
+        if sleep_duration >= MAX_SLEEP_DURATION
+          sleep_duration = MAX_SLEEP_DURATION
+        else
+          sleep_duration = sleep_duration * 2
```

Comment:
super minor but technically you are allowing sleep duration to extend beyond max. we should do the clamping after the multiplication I guess.